src/Raven.Server/Documents/Handlers/Processors/Streaming/StreamingHandlerProcessorForGetStreamQuery.cs

```diff
@@ -0,0 +1,146 @@
+            }
+        }
+
+        protected override IDisposable AllocateContext(out DocumentsOperationContext context)
+        {
+            var queryContext = QueryOperationContext.Allocate(RequestHandler.Database);
+            context = queryContext.Documents;
+            _queryContext = queryContext;
+            return queryContext;
+        }
+
+        protected override QueryMetadataCache GetQueryMetadataCache()
+        {
+            return RequestHandler.Database.QueryMetadataCache;
+        }
+
+        protected override async ValueTask ExecuteAndWriteIndexQueryStreamEntriesAsync(DocumentsOperationContext context, IndexQueryServerSide query, string format, string _,
+            string[] propertiesArray, string fileNamePrefix, bool ignoreLimit, bool fromSharded, OperationCancelToken token)
+        {
+            //writes either csv or blittable documents if is shard
+            await using (var writer = GetBlittableQueryResultWriter(format, isDebug: true, context, HttpContext.Response, RequestHandler.ResponseBodyStream(), fromSharded, propertiesArray, fileNamePrefix))
+            {
+                try
+                {
+                    await RequestHandler.Database.QueryRunner.ExecuteStreamIndexEntriesQuery(query, _queryContext, HttpContext.Response, writer, ignoreLimit, token).ConfigureAwait(false);
+                }
+                catch (IndexDoesNotExistException)
+                {
+                    HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
+                    await writer.WriteErrorAsync($"Index {query.Metadata.IndexName} does not exist");
+
+                    if (fromSharded)
+                        throw;
+                }
+            }
+        }
+
+        protected override async ValueTask ExecuteAndWriteQueryStreamAsync(DocumentsOperationContext context, IndexQueryServerSide query, string format,
+            string[] propertiesArray, string fileNamePrefix, bool _, bool fromSharded, OperationCancelToken token)
+        {
+            await using (var writer = GetDocumentQueryResultWriter(format, HttpContext.Response, context, RequestHandler.ResponseBodyStream(), propertiesArray,
+                             fileNamePrefix))
+            {
+                try
+                {
+                    await RequestHandler.Database.QueryRunner.ExecuteStreamQuery(query, _queryContext, HttpContext.Response, writer, token).ConfigureAwait(false);
+                }
+                catch (IndexDoesNotExistException)
+                {
+                    HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
+                    await writer.WriteErrorAsync($"Index {query.Metadata.IndexName} does not exist");
+
+                    if (fromSharded)
+                        throw;
+                }
+                catch (Exception e)
+                {
+                    if (_method == HttpMethod.Get)
```

Comment:
Do you know why we have it only for Get?

Reply:
This was originally like this, not sure why. IIRC all of the client API stream query funcs use `POST` so I suspect the `GET` is only used manually (or from studio) and only there we would want to write the error to stream

Reply:
ok